PICMI_Python/base.py

```diff
@@ -32,7 +32,23 @@ def register_constants(implementation_constants):
 def _get_constants():
     return _implementation_constants
 
-class _ClassWithInit(object):
+
+class _DocumentedMetaClass(type):
+    """This is used as a metaclass that combines the __doc__ of the picmistandard base and of the implementation"""
+    def __new__(cls, name, bases, attrs):
+        # "if bases" skips this for the _ClassWithInit (which has no bases)
+        # "if bases[0].__doc__ is not None" skips this for the picmistandard classes since their bases[0] (i.e. _ClassWithInit)
+        # has no __doc__.
+        if bases and bases[0].__doc__ is not None:
+            implementation_doc = attrs.get('__doc__', '')
+            if implementation_doc:
+                attrs['__doc__'] = bases[0].__doc__ + """\n\n    Implementation specific documentation\n""" + implementation_doc
```

Comment:
Are the spaces here intentional/generally the right indentation?

Reply:
The spaces are intentional. The double return `\n\n` is needed to start a new section. Then the four spaces indent to the same level as a standard doc string (assuming PEP8 indentation). I'll add a comment to that effect.